main.go

```diff
@@ -92,9 +93,17 @@ func runDaemon() error {
 	if _, err := os.Stat(cfg.ContainerdAddr); err == nil {
 		os.RemoveAll(cfg.ContainerdAddr)
 	}
+
+	containerdBinary := "containerd"
+	if cfg.ContainerdPath != "" {
+		containerdBinary = cfg.ContainerdPath
+	}
+	if _, err := osexec.LookPath(containerdBinary); err != nil {
```

Comment:
I do not you did `find containerd binary in linux path first`.

`func LookPath(file string) (string, error)` searches for an executable binary named `file` in the directories named by the PATH environment variable.

But here `containerdBinary ` can be a file name and also can be a path string.

Reply:
it won't be found by `LookPath` if  `containerBinary` is just a file name
https://golang.org/src/os/exec/lp_unix.go?s=931:973#L24

Reply:
I updated , make variable named more readable, PTAL